src/base.jl

```diff
@@ -48,21 +45,45 @@ size_nibbles(::Type{T}) where T = ceil(log2(sizeof(T))/4)
 
 size_mask(T) = UInt(exp2(4*size_nibbles(T)) - 1)
 
+function Base.getindex(s::ShortString, i::Integer)
+     getindex(String(s), i)
+end
 
-# function Base.getindex(s::ShortString, i::Integer)
-#     getindex(String(s), i)
-# end
+function Base.getindex(s::ShortString, args...; kwargs...)
+     getindex(String(s), args...; kwargs...)
+end
 
-# function Base.getindex(s::ShortString, args...; kwargs...)
-#     getindex(String(s), args...; kwargs...)
-# end
 
-Base.collect(s::ShortString) = collect(String(s))
+function Base.iterate(s::ShortString{T}, i::Integer=1) where T
+    i > ncodeunits(s) && return nothing
+    shifted = s.size_content >> 8*(sizeof(T) - i)
+    if shifted % UInt8 <= 0x7f  # 1 byte character
+        return Char(shifted % UInt8), i+1
+    elseif shifted % UInt16 <= 0x7ff  # 2 byte character
+        return Char(shifted % UInt16 % UInt32), i+2
+    elseif shifted % UInt32 <= 0xffff  # 3 byte character
+        return Char(shifted % UInt32 & 0xffff), i+3
+    else  # 4 byte character
+        return Char(shifted % UInt32), i+3
+    end
```

Comment:
I am not sure i have this right

Reply:
Yep it is wrong.
to do this right Need to look at 4 byte chunks.

Reply:
Do you want me to write this function?

Reply:
yes that would be useful.

Reply:
```
@inline _get_word(s::ShortString{T}, i::Int) where {T} =
           (x.size_content >> (8*(sizeof(T) - i - 3)))%UInt32

@inline function Base.iterate(s::ShortString, i::Int=1)
       0 < i <= ncodeunits(s) || return nothing
       chr = _get_word(s, i)
       chr < 0x8000_0000 ? (reinterpret(Char, chr & 0xFF00_0000), i + 1) :
           chr < 0xe000_0000 ? (reinterpret(Char, chr & 0xFFFF_0000), i + 2) :
           chr < 0xf000_0000 ? (reinterpret(Char, chr & 0xFFFF_FF00), i + 3) :
           (reinterpret(Char, chr), i + 4)
end
```
Try this

Reply:
I think that's probably about as fast as possible for this, since it keeps the bytes in the same big endian order that both ShortString and Char use, and just does nothing more than a shift and mask

Reply:
I don't know how `reinterpret` works here for 4 byte characters
https://discourse.julialang.org/t/how-does-char-get-stored/49366

Reply:
I split out just the fast iterate code above, and put it into a separate PR, I hope you don't mind, @oxinabox!
I'm just trying to get things moving along as quickly as possible 😄